SOURCES/etc/xapi.d/plugins/raid.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from functools import wraps
+import sys
+import json
+
+import XenAPIPlugin
+
+sys.path.append('.')
+from xcpngutils import configure_logging, run_command
+from xcpngutils.filelocker import FileLocker
+
+_LOGGER = configure_logging('raid')
+
+
+def lock(*pid_args, **pid_kwargs):
+    def wrapper(func):
+        @wraps(func)
+        def decorator(*args, **kwargs):
+            try:
+                with FileLocker(*pid_args, **pid_kwargs):
+                    return func(*args, **kwargs)
+            except Exception as e:
+                return json.dumps({'error': str(e)})
+        return decorator
+    return wrapper
+
+
+# returns {"status": {"State": "clean", (...)}, "volumes": [["0", "8", "0", "0", "active sync", "/dev/sda"], (...)]}
+@lock()
+def check_raid_pool(session, args):
+    device = '/dev/md127'
```

Comment:
We'd check only this device? What if the users created other raids and would like to monitor them?

Reply:
Right now the main objective is to check the system RAID created during install. Maybe later, we'll have a more advanced RAID detection capabilities, but it's not the main objective for now